frame/pablo/src/lib.rs

```diff
@@ -573,6 +634,107 @@ pub mod pallet {
 	}
 
 	impl<T: Config> Pallet<T> {
+		fn default_pblo_staking_pool_config(
+			pool_id: &T::PoolId,
+			pair: &CurrencyPair<T::AssetId>,
+		) -> Result<
+			RewardPoolConfiguration<
+				T::AccountId,
+				T::AssetId,
+				T::BlockNumber,
+				RewardConfigsOf<T>,
+				DurationPresets<T>,
+			>,
+			DispatchError,
+		> {
+			let mut rewards = BTreeMap::new();
+			let max_rewards: T::Balance = T::Convert::convert(MAX_REWARDS);
+			let reward_rate = Perbill::from_percent(REWARD_PERCENTAGE);
+			rewards
+				.insert(pair.base, RewardConfig { asset_id: pair.base, max_rewards, reward_rate });
+			rewards.insert(
+				pair.quote,
+				RewardConfig { asset_id: pair.quote, max_rewards, reward_rate },
+			);
+			let reward_configs = RewardConfigsOf::<T>::try_from(rewards)
+				.map_err(|_| Error::<T>::StakingPoolConfigError)?;
+			let mut duration_presets = BTreeMap::new();
+			duration_presets.insert(ONE_HOUR, Perbill::from_percent(1));
+			duration_presets.insert(ONE_MINUTE, Perbill::from_percent(10));
+			let duration_presets = BoundedBTreeMap::try_from(duration_presets)
+				.map_err(|_| Error::<T>::StakingPoolConfigError)?;
+			let lock = LockConfig { duration_presets, unlock_penalty: Perbill::from_percent(5) };
+			Ok(RewardPoolConfiguration::<_, _, _, _, _>::RewardRateBasedIncentive {
+				owner: Self::account_id(pool_id),
+				asset_id: primitives::currency::CurrencyId::PBLO.0.into(),
+				end_block: frame_system::Pallet::<T>::current_block_number() +
+					FIVE_YEAR_BLOCKS.into(),
+				reward_configs,
+				lock,
+			})
```

Comment:
Consider using `[foo, bar].into_iter().try_collect()` instead of `mut` and `.insert`